<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>865</width>
    <height>607</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>UDP MIDI Test Application</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>NIC</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="cbNic"/>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Destination IP:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="leTargetIp">
        <property name="text">
         <string>127.0.0.1</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Port</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="sbTargetPort">
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>64116</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="cbMidiOut"/>
      </item>
      <item>
       <widget class="QCheckBox" name="cbPlayRx">
        <property name="text">
         <string>Play RX</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbPlayTx">
        <property name="text">
         <string>Play TX</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnStart">
        <property name="text">
         <string>Start</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Transmit</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Sent Messages</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="QPlainTextEdit" name="teBuffer"/>
           </item>
           <item row="1" column="0">
            <widget class="QTabWidget" name="tabWidget_2">
             <property name="currentIndex">
              <number>1</number>
             </property>
             <widget class="QWidget" name="tab_3">
              <attribute name="title">
               <string>Notes</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <item>
                <widget class="PianoKeybd" name="keyboardWidget" native="true">
                 <property name="minimumSize">
                  <size>
                   <width>0</width>
                   <height>50</height>
                  </size>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tab_4">
              <attribute name="title">
               <string>Show Control</string>
              </attribute>
              <layout class="QGridLayout" name="gridLayout_2">
               <item row="0" column="1" colspan="2">
                <widget class="QSpinBox" name="sbMSCDevId">
                 <property name="maximum">
                  <number>255</number>
                 </property>
                 <property name="value">
                  <number>1</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>Command Format</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1" colspan="2">
                <widget class="QComboBox" name="cbMSCCommandFormat"/>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>Command</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1" colspan="2">
                <widget class="QComboBox" name="cbMSCCommand"/>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>Command Data</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1" colspan="2">
                <widget class="QLineEdit" name="leMSCData"/>
               </item>
               <item row="3" column="3">
                <widget class="QComboBox" name="cbMscDataType">
                 <item>
                  <property name="text">
                   <string>Hex Bytes</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Eos Cue Format</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>Device ID</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="1">
                <widget class="QLineEdit" name="leMSCComposedData">
                 <property name="styleSheet">
                  <string notr="true">background-color: rgb(190, 190, 190);
font: 8pt &quot;Courier New&quot;;</string>
                 </property>
                 <property name="text">
                  <string>80 C0 F0 D0 F0 0F</string>
                 </property>
                 <property name="readOnly">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="5" column="3">
                <widget class="QPushButton" name="btnMSCSend">
                 <property name="text">
                  <string>Send</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="0">
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>Composed Message</string>
                 </property>
                </widget>
               </item>
               <item row="5" column="1">
                <widget class="QLabel" name="lbInvalidData">
                 <property name="text">
                  <string>Invalid Data</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Recieve</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Timecode</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLCDNumber" name="nTimecode">
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>100</height>
              </size>
             </property>
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="smallDecimalPoint">
              <bool>false</bool>
             </property>
             <property name="digitCount">
              <number>11</number>
             </property>
             <property name="mode">
              <enum>QLCDNumber::Dec</enum>
             </property>
             <property name="segmentStyle">
              <enum>QLCDNumber::Flat</enum>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="cbLogAllInput">
          <property name="text">
           <string>Log All Messages</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="lvRxMessages"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>865</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PianoKeybd</class>
   <extends>QWidget</extends>
   <header>vkey/pianokeybd.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
